src/nvim/ex_docmd.c

```diff
@@ -9625,20 +9625,34 @@ static void ex_folddo(exarg_T *eap)
 static void ex_terminal(exarg_T *eap)
 {
   char *name = (char *)p_sh;  // Default to 'shell' if {cmd} is not given.
+  char ex_cmd[1024] = { 0 };
```

Comment:
We don't need to initialize `ex_cmd`. Both `if` branches start filling the array from position `0`.